src/persist-client/src/read.rs

```diff
@@ -570,36 +588,39 @@ where
             }
         };
     };
-
-    let batch = BlobTraceBatchPart::decode(&value)
-        .map_err(|err| anyhow!("couldn't decode batch at key {}: {}", key, err))
-        // We received a State that we couldn't decode. This could
-        // happen if persist messes up backward/forward compatibility,
-        // if the durable data was corrupted, or if operations messes up
-        // deployment. In any case, fail loudly.
-        .expect("internal error: invalid encoded state");
-    let mut ret = Vec::new();
-    for chunk in batch.updates {
-        for ((k, v), t, d) in chunk.iter() {
-            let mut t = T::decode(t);
-            if !desc.lower().less_equal(&t) {
-                continue;
-            }
-            if desc.upper().less_equal(&t) {
-                continue;
-            }
-            // NB: t_fn might mutate t (i.e. a snapshots will forward it to the
-            // as_of), but the desc checks above only make sense if it hasn't be
-            // mutated yet.
-            let keep = t_fn(&mut t);
-            if !keep {
-                continue;
+    drop(get_span);
+
+    trace_span!("fetch_batch::decode").in_scope(|| {
+        let batch = BlobTraceBatchPart::decode(&value)
```

Comment:
oh nice, this block is cpu-bound work!

Reply:
yup! and it's significant enough at high throughputs that it's worthwhile to see on the UI